git-github-introduction.md

```diff
@@ -0,0 +1,340 @@
+ * branch            master     -> FETCH_HEAD
+  * [new branch]      master     -> upstream/master
+  Already up-to-date.
+```
+
+### Add a New Branch Locally
+
+First, list all known local branches of the repository, by typing the following:  
+
+```
+$ git branch
+* master
+```
+
+Next, to add a new branch, which is the preferred method for working on and adding new features, type **git branch new_branch**, where **''new_branch**'' is the name of the new branch:
+
+```
+$ git branch new_branch
+```
+
+The next step is to switch to the newly created branch:
+
+```
+$ git checkout new_branch
+Switched to branch 'new_branch'
+```
+
+
+### Commit a New File to the Branch
+
+Create a `README.txt` file and save it to the local machine. You do this by opening a new document in a text editor and saving as `README.txt` in the current directory. In the README file you would typically put instructions or details about the project. 
+
+The next step is to ask Git if it can find the README.txt file, by asking it to identify all untracked files (on the local repository):
+
+```
+$ git status
+On branch new_branch
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+
+    README.txt
+
+    nothing added to commit but untracked files present (use "git add" to track)
+```
+
+Git indicates that the README.txt file is untracked, so the first command (git add) will add it. The second command (git status) will tell us whether the file has been added successfully. 
+
+```
+$ git add README.txt
+
+$ git status
+On branch new_feature
+Changes to be committed:
+  (use "git reset HEAD <file>..." to unstage)
+
+    new file:   README.txt
+```
+
+After adding the file it must be committed. By committing a file, Git creates a message about the file that will be transmitted once the file is ready to be pushed to the master branch.  A Git message should be descriptive and mention the reason for the change, or changes, to be committed.  In the following example, the `-m` flag (for message) is used and a commit message is appended in quotes.
+
+```
+$ git commit -m "Add a README file to the project"
+[new_branch 7660f34] Add a README file to the project
+ 1 file changed, 1 insertion(+)
+  create mode 100644 README.txt
+```
+
+If you want to check to see that all files have been added and committed, type **git status** again. The status should show that the directory is clean, as in the example below:
+
+```
+$ git status
+On branch new_branch
+nothing to commit, working directory clean
+```
+
+To get a list of commit messages made thus far, type **git log**. 
+This will show the user's information and the commit message (in this case "Add a README to the project"):
+
+```
+$ git log
+commit 3c9b1d0068075fdac6ed928f8e5a27ac3253ca5e
+Author: John Smith <john.smith@example.com>
+Date:   Tue Sep 2 14:46:36 2014 -0400
+
+    Add a README to the project.
+```
+
+
+### Push the Commit to the Remote (Forked) Repository
+

```

Comment:
Further above we call *upstream* the "forked" repo. I think here you are referring to "origin". I think we should establish/use common terminology, e.g.:
- **SeattleTestbed/repy_v2**:  upstream, base repo, original repo, repo that or from which was forked
- **vladimir-v-diaz/repy_v2**: origin, fork, copy